src/libcrun/handlers/wasmedge.c

```diff
@@ -169,14 +169,40 @@ static int
 wasmedge_can_handle_container (libcrun_container_t *container, libcrun_error_t *err arg_unused)
 {
   const char *annotation;
+  const char *entrypoint_executable = container->container_def->process->args[0];
 
   annotation = find_annotation (container, "run.oci.handler");
   if (annotation)
-    return strcmp (annotation, "wasm") == 0 ? 1 : 0;
+    {
+
+      /* wasm-smart: annotation is a smart switch which only toggles wasm if its necessary,
+         following annotation is very useful for cases where users intends to run wasm workload on
+         kubernetes cluster but workload also contains side-cars which could execute non-wasm worload.
+         Example: Kubernetes clusters with service-mesh such as istio, linkerd etc
+      */
+      if (strcmp (annotation, "wasm-smart") == 0)
```

Comment:
Would it make sense to make a new function that took either "wasm" or "compat" rather then copying this code around 4 times?

Reply:
fixed in new commit.